server/message_queue_service.py

```diff
@@ -0,0 +1,126 @@
+import asyncio
+import json
+from typing import Dict
+
+import aio_pika
+from aio_pika import DeliveryMode, ExchangeType
+from aio_pika.exceptions import ProbableAuthenticationError
+
+from .config import config, TRACE
+from .core import Service
+from .decorators import with_logger
+
+
+@with_logger
+class MessageQueueService(Service):
+    def __init__(self) -> None:
+        """
+        Service handling connection to the message queue
+        and providing an interface to publish messages.
+        """
+        self._logger.debug("Message queue service created.")
+        self._connection = None
+        self._channel = None
+        self._exchanges = {}
+        self._exchange_types = {}
+
+        config.register_callback("MQ_USER", self.reconnect)
+        config.register_callback("MQ_PASSWORD", self.reconnect)
+        config.register_callback("MQ_VHOST", self.reconnect)
+        config.register_callback("MQ_PORT", self.reconnect)
+
+    async def initialize(self) -> None:
+        if self._connection is not None:
+            return
+
+        await self._connect()
+
+    async def _connect(self) -> None:
+        try:
+            self._connection = await aio_pika.connect_robust(
+                "amqp://{user}:{password}@localhost:{port}/{vhost}".format(
+                    user=config.MQ_USER,
+                    password=config.MQ_PASSWORD,
+                    vhost=config.MQ_VHOST,
+                    port=config.MQ_PORT,
+                ),
+                loop=asyncio.get_running_loop(),
+            )
+        except ConnectionError:
+            self._logger.warning("Unable to connect to RabbitMQ. Is it running?")
```

Comment:
I would include the exception trace here so it's obvious something went wrong. Otherwise it's really easy to miss this warning.